### DISCUSSION TITLE
Using urlrewrite to grab download page and then parse it

### DISCUSSION THREAD

**Discussion by art-losangeles** [Flexget/Flexget#3487] [Q&A]
I am on a site that has an RSS feed that gives a link to a page for each new torrent (a "viewtopic" page). The link has no download ID so cannot be rewritten directly. But, the "viewtopic page has a download link. I see multiple places that say I can grab the torrent page (the "viewtopic" page in my case) and then process the download link from that page but I cannot find an example of how to do that. All I see is how to grab the ID from the link to the new torrent page (the "viewtopic" page here) and then changing that into a download link directly, using the same ID number. But, in my case the numerical IDs are different so that wont work.

Is it possible to do what I want? Or, perhaps the descriptions that say I can do that are wrong. If I can do that, how do I do it?

Site is 720pier.ru
RSS links are in the form of "720pier.ru/viewtopic.php?t=51193" [real example; note "viewtopic.php?t="]
The download link on the viewtopic page is in the form of "720pier.ru/download/torrent?id=50735" [download link from above; note a different 5-digit number and "download/torrent?id=" instead of the "veiwtopic.php?t=" link]
On each "viewtopic" page there is only one download link so that shouldn't be a problem. Checking a bunch of links there is no fixed relation between the viewtopic ID and the download ID so I'm going to need to actually grab the download link from the viewtopic page.

Help greatly appreciated. I'm a newbie with urlwriter plugin so maybe I actually need another plugin as well, such as the html plugin. Unfortunately, I think I'm going to need a detailed explanation but hopefully that already exists somewhere and link will be sufficient.

Art in Los Angeles

**Comment by lamski** [comment]
Try reading the `html` and `manipulate` plugins from https://flexget.com/Plugins
That's what I use to parse raw HTML and extract download links from it.

**Comment by bencheg** [comment]
[art-losangeles](https://github.com/art-losangeles), you found a way to get flexget to click on links?

**Reply by art-losangeles** [reply]
No, I gave up. I might have figured it out with a working example of how the HTML plugin works but I didn't found any such thing.

**Comment by gazpachoking** [comment]
I think with our current tools you'd have to write a custom urlrewriter plugin that downloaded the intermediate page and did the parsing itself.
